coderd/dormancy/dormantusersjob_test.go

```diff
@@ -0,0 +1,92 @@
+package dormancy_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/moby/moby/pkg/namesgenerator"
+	"github.com/stretchr/testify/require"
+
+	"cdr.dev/slog/sloggers/slogtest"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/database/dbfake"
+	"github.com/coder/coder/coderd/dormancy"
+	"github.com/coder/coder/testutil"
+)
+
+func TestCheckInactiveUsers(t *testing.T) {
+	t.Parallel()
+
+	// Predefine job settings
+	interval := time.Millisecond
+	dormancyPeriod := 90 * 24 * time.Hour
+
+	// Add some dormant accounts
+	logger := slogtest.Make(t, &slogtest.Options{IgnoreErrors: true})
+	db := dbfake.New()
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	inactiveUser1 := setupUser(ctx, t, db, "dormant-user-1@coder.com", time.Now().Add(-dormancyPeriod).Add(-time.Minute))
+	inactiveUser2 := setupUser(ctx, t, db, "dormant-user-2@coder.com", time.Now().Add(-dormancyPeriod).Add(-time.Hour))
+	inactiveUser3 := setupUser(ctx, t, db, "dormant-user-3@coder.com", time.Now().Add(-dormancyPeriod).Add(-6*time.Hour))
+
+	activeUser1 := setupUser(ctx, t, db, "active-user-1@coder.com", time.Now().Add(-dormancyPeriod).Add(time.Minute))
+	activeUser2 := setupUser(ctx, t, db, "active-user-2@coder.com", time.Now().Add(-dormancyPeriod).Add(time.Hour))
+	activeUser3 := setupUser(ctx, t, db, "active-user-3@coder.com", time.Now().Add(-dormancyPeriod).Add(6*time.Hour))
+
+	// Run the periodic job
+	closeFunc := dormancy.CheckInactiveUsersWithOptions(ctx, logger, db, interval, dormancyPeriod)
+	t.Cleanup(closeFunc)
+
+	var rows []database.GetUsersRow
+	var err error
+	require.Eventually(t, func() bool {
+		rows, err = db.GetUsers(ctx, database.GetUsersParams{})
+		if err != nil {
+			return false
+		}
+
+		var c int
+		for _, row := range rows {
+			if row.Status == database.UserStatusDormant {
+				c++
+			}
+		}
+		// 6 users in total, 3 dormant
+		return len(rows) == 6 && c == 3
+	}, testutil.WaitShort, testutil.IntervalMedium)
+
```

Comment:
Suggestion: if you allow passing in a `tick chan time.Time` and `updated chan int` into `CheckInactiveUsersWithOptions` you can then conceivably remove this `require.Eventually` - you write a time to `tick`, and read the number of updated users from `updated`

Reply:
This is a good hint in case we want to do something more with these accounts than validate in unit tests. I would keep it simple as is.